libraries/AP_HAL_SITL/GPIO.cpp

```diff
@@ -50,7 +49,7 @@ void GPIO::toggle(uint8_t pin)
 
 /* Alternative interface: */
 AP_HAL::DigitalSource* GPIO::channel(uint16_t n) {
-    return new DigitalSource(0);
+    return new DigitalSource(0);  // TODO : should be n ?
```

Comment:
yes, should be n, should also return nullptr if it is >= 8 (ie. sizeof(pin_mask)*8)